File: Semester 1/Imperative Programming/lab3/pascal.py
import math

n = int(input())

for i in range(n+1):
    #print(" " * (math.ceil(n / 2) - i), end="")
    for j in range(i+1):
        if j == 0:
            print("1 ", end="")
        else:
            print(str(int(math.factorial(i) / (math.factorial(j)*math.factorial(i - j)))) + " ", end="")
    print()

File: Semester 1/Imperative Programming/lab3/pascal.c
#include <stdio.h>
#include <math.h>

long factorial(int n);

int main(void)
{
    int n;
    scanf("%d", &n);

    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n - 1; j++)
        {
            printf(" ");
        }

        for (int j = 0; j < i; j++)
        {
            printf("%ld ", factorial(i) / (factorial(j) * factorial(i - j)));
        }

        printf("\n");
    }

    return 0;
}

long factorial(int n)
{
  if (n == 0)
  {
    return 1;  
  }
  else
  {
        return(n * factorial(n-1));  
  }
}